perses/rjmc/geometry.py

```diff
@@ -650,10 +646,15 @@ def _calculate_angle(self, atom_position, bond_atom_position, angle_atom_positio
         b = angle_atom_position - bond_atom_position
         a_u = a / np.linalg.norm(a)
         b_u = b / np.linalg.norm(b)
-        theta = np.arccos(np.dot(a_u, b_u))
+        cos_theta = np.dot(a_u, b_u)
```

Comment:
This is the bug.  Because of the way you define `a` and `b` vectors, this should be
```
cos_theta = np.dot(-a_u, b_u)
```